### PR TITLE
Use MK compiled for windows/amd64

### PR DESCRIPTION
**Pull request by bassosimone**
- update examples to not assume a unix path for cert.pem

- remove `i/m/engine.h`: the proper header should be `ffi.h` so I
  am not sure how the code could work, maybe there was an header
  in some well know system path?

- point to the subrepository version with the DLLs (see https://github.com/measurement-kit/libs/pull/1)

- tweak compiler flags in measurement_kit.go

Some notes:

- as such, this branch probably breaks the macOS build, so more
  work is probably needed to integrate all smoothly

- `measurement_kit.dll` must be copied in the same directory
  of the `.exe` files for them to work correctly with it

- the resulting binaries still link `msvcrt.dll` but since
  `measurement_kit.dll` is statically linked with the static
  multi-threaded C runtime, there should be no issue

- `cert.pem` must now be in the current directory of the `.exe`

### CODE REVIEW

File: include/measurement_kit/engine.h [hunk 1]
```diff
@@ -1,194 +0,0 @@
...
-    XX(HttpInvalidRequestLine)                                                 \
-    XX(MeekFrontedRequests)                                                    \
-    XX(MultiNdt)                                                               \
-    XX(Ndt)                                                                    \
-    XX(TcpConnect)                                                             \
-    XX(Telegram)                                                               \
-    XX(WebConnectivity)                                                        \
-    XX(Whatsapp)
-
-#endif /* MEASUREMENT_KIT_ENGINE_H */
```
Comment (hellais) [inline include/measurement_kit/engine.h]: Why did you delete this?

File: measurement_kit.go [hunk 2]
```diff
@@ -3,15 +3,18 @@ package mk
...
+// #cgo darwin CFLAGS: -I${SRCDIR}/libs/include
+// #cgo darwin LDFLAGS: -lc++
 // #cgo darwin,amd64 LDFLAGS: ${SRCDIR}/libs/libevent-darwin-amd64.a
 // #cgo darwin,amd64 LDFLAGS: ${SRCDIR}/libs/libevent_openssl-darwin-amd64.a
 // #cgo darwin,amd64 LDFLAGS: ${SRCDIR}/libs/libevent_pthreads-darwin-amd64.a
 // #cgo darwin,amd64 LDFLAGS: ${SRCDIR}/libs/libcrypto-darwin-amd64.a
 // #cgo darwin,amd64 LDFLAGS: ${SRCDIR}/libs/libssl-darwin-amd64.a
 // #cgo darwin,amd64 LDFLAGS: ${SRCDIR}/libs/libGeoIP-darwin-amd64.a
 // #cgo darwin,amd64 LDFLAGS: ${SRCDIR}/libs/libmeasurement_kit-darwin-amd64.a
+// #cgo windows CFLAGS: -I${SRCDIR}/libs/include
```
Comment (hellais) [inline measurement_kit.go]: You don't have to duplicate this since it's the same for macOS, linux and windows.

**Comment by hellais** [comment]
Good stuff!

I will test this more carefully on monday, but some things that popped to my mind.

* I would keep the `include` directory committed to this repo and not in the subrepo. The point of having the subrepo is just to avoid bloating this with too big files. Since the headers are quite compact I would keep them here.

* The `cert.pem` path should probably be determined based on some helper function, but I would keep it, for the sake of example, in the standard unix path and maybe write a comment next to it saying that it's advisable to use a helper in real code (since that path will "just work" for most users cloning the go-measurement-kit repo).

File: measurement_kit.go (hunk 2 above)
Comment (bassosimone) [reply measurement_kit.go]: Yeah, I should have unified flags in 84e6051

**Comment by bassosimone** [comment]
> I would keep the include directory committed to this repo and not in the subrepo.

My objective was to put in the _same_ place the artifacts and the header. In the event in which they are out of sync, it's an issue. But maybe `golang` should query at runtime the version number and make sure it is compatible with the version number that it expects.

**Comment by bassosimone** [comment]
> The cert.pem path should probably be determined based on some helper function

Maybe make it dependent on the system? Also the `CA` is an asset, maybe `s/libs/assets`?

**Comment by hellais** [comment]
> Maybe make it dependent on the system? Also the CA is an asset, maybe s/libs/assets?

On second though, that should probably be the responsibility of the user of the library to pass in a correct `cert.pem`. Since these are example codes, they are meant to illustrate a point and as such it's OK that the path will not work on every single platform.

I am going to put back the original paths.

**Comment by hellais** [comment]
> My objective was to put in the same place the artifacts and the header. In the event in which they are out of sync, it's an issue. But maybe golang should query at runtime the version number and make sure it is compatible with the version number that it expects.


Yes, I understand that. However the point of putting things in a submodule is just to avoid committing too big files in the master repo. The version of the subrepo is still checked out so the out of sync issues you mention should not happen when using this repo.

**Comment by bassosimone** [comment]
> I am going to put back the original paths.

Please, don't. Consider that we most likely want to have CI in this repo. What would be golden would be to build on both Windows (with appveyor) and Linux (travis) and run smoke tests. In such scenario, having the examples actually working (rather than being dependent on magic) seems to be better.

**Comment by bassosimone** [comment]
> The version of the subrepo is still checked out so the out of sync issues you mention should not happen when using this repo.

I would recommend not doing that. What can break will break. Better to have a design where it's less likely that things break without us noticing, no?

**Comment by bassosimone** [comment]
Superseded by #6.

**Comment by bassosimone** [comment]
Closing since #6 now works on macOS.
